Lib/fontParts/base/point.py

```diff
@@ -1,19 +1,23 @@
-from numbers import Number
-from typing import List, Optional
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, List, Optional, Tuple
 
 from fontParts.base import normalizers
 from fontParts.base.annotations import TransformationMatrixType
 from fontParts.base.base import (BaseObject, IdentifierMixin,
                                  PointPositionMixin, SelectionMixin,
                                  TransformationMixin, dynamicProperty,
                                  reference)
-from fontParts.base.contour import BaseContour
-from fontParts.base.deprecated import DeprecatedPoint, RemovedPoint
-from fontParts.base.font import BaseFont
-from fontParts.base.glyph import BaseGlyph
-from fontParts.base.layer import BaseLayer
 from fontTools.misc import transform
 
+if TYPE_CHECKING:
+    from fontParts.base.annotations import IntFloatType
+    from fontParts.base.contour import BaseContour
+    from fontParts.base.deprecated import DeprecatedPoint, RemovedPoint
```

Comment:
`DeprecatedPoint` and `RemovedPoint` are `BasePoint` base classes that need to be imported at runtime, an thus should not be included in the `TYPE_CHECKING` conditional.